src/WorkflowCore/Services/WorkflowExecutor.cs

```diff
@@ -141,6 +141,21 @@ private async Task ExecuteStep(WorkflowInstance workflow, WorkflowStep step, Exe
             {
                 _logger.LogDebug("Starting step {0} on workflow {1}", step.Name, workflow.Id);
 
+                IStepExecutionContext context = new StepExecutionContext()
+                {
+                    Workflow = workflow,
+                    Step = step,
+                    PersistenceData = pointer.PersistenceData,
+                    ExecutionPointer = pointer,
+                    Item = pointer.ContextItem
+                };
+
+                if (scope.ServiceProvider.GetService<IStepExecutionContextAccessor>()
+                    is StepExecutionContextAccessor accessor)
+                {
+                    accessor.StepExecutionContext = context;
```

Comment:
I think, instead of baking this into the core library... I would put an event that the user can hook into here... something like `OnExecutionScopeConstruction` or something like that.  Then the user can customize the execution scope to have whatever they want.

Reply:
In that case, why not introduce some sort of StepExecutionFilter; a way of adding code that will run before and\or after every Step within one or multiple Workflows? Similar to Filter\Middleware in ASP.NET or Pipeline-behavior i Mediatr. This is far beyond what this branch was trying to solve, but would be a very powerful feature for implementing cross-cutting concerns.

Reply:
Sorry, I missed this update... that sounds like an interesting idea!

Reply:
Yes,  I agree Middleware would be a great feature to have